dartsim/src/SDFFeatures.cc

```diff
@@ -773,8 +773,18 @@ Identity SDFFeatures::ConstructSdfCollision(
     }
     if (odeFriction->HasElement("fdir1"))
     {
-      math::Vector3d fdir1 = odeFriction->Get<math::Vector3d>("fdir1");
+      auto frictionDirectionElem = odeFriction->GetElement("fdir1");
+      math::Vector3d fdir1 = frictionDirectionElem->Get<math::Vector3d>();
       aspect->setFirstFrictionDirection(math::eigen3::convert(fdir1));
+
+      const std::string kExpressedIn = "ignition:expressed_in";
+      if (frictionDirectionElem->HasAttribute(kExpressedIn))
+      {
+        auto skeleton = bn->getSkeleton();
+        auto directionFrameBodyNode = skeleton->getBodyNode(
```

Comment:
Should we check if the `BodyNode` exists and print an error if it doesn't?

Reply:
yeah, that's a reasonable idea

Reply:
6f61a92